vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/experiments/experiment/components/ExistingGroupsPreventVariableEdit.java

```diff
@@ -0,0 +1,53 @@
+package life.qbic.datamanager.views.projects.project.experiments.experiment.components;
+
+import com.vaadin.flow.component.Text;
+import com.vaadin.flow.component.button.Button;
+import com.vaadin.flow.component.html.Div;
+import com.vaadin.flow.component.html.Span;
+import com.vaadin.flow.component.icon.Icon;
+import com.vaadin.flow.component.icon.VaadinIcon;
+import life.qbic.datamanager.views.notifications.NotificationDialog;
+
+/**
+ * Notifies the user that experimental groups exist in the experiment.
+ * <p>
+ * This dialog is to be shown when editing variables is impossible as experimental groups are
+ * present.
+ */
+public final class ExistingGroupsPreventVariableEdit extends NotificationDialog {
```

Comment:
NIT: I think having the component type in the name helps understand what it is for within the structure tree 🤔
```suggestion
public final class ExistingGroupsPreventVariableEditDialog extends NotificationDialog {
```

Reply:
I think this is clear from the extension and not needed here.